kerastuner/engine/base_tuner.py

```diff
@@ -163,6 +163,38 @@ def run_trial(self, trial, x, y, val_x, val_y):
         """
         raise NotImplementedError
 
+    def _run_fit(self, model, hp, *args, **kwargs):
```

Comment:
Let's put this in Tuner, not BaseTuner, since BaseTuner is supposed to be generic and work for non-Keras models